sqlglot/expressions.py

```diff
@@ -1439,6 +1439,11 @@ class Describe(Expression):
     arg_types = {"this": True, "style": False, "kind": False, "expressions": False}
 
 
+# https://duckdb.org/docs/guides/meta/summarize.html
+class Summarize(Expression):
+    arg_types = {"this": True, "table": False}
```

Comment:
Just out of curiosity why is `table` an argument?

`TABLE t` is equivalent to `SELECT * FROM t` and summarize works like

`SUMMARIZE {any valid row-returning statement}`

and `{any valid row-returning statement}` would include `TABLE t`.

Reply:
Because the docs have an [example](https://duckdb.org/docs/guides/meta/summarize.html#summarizing-remote-tables) that doesn't work without the `TABLE` keyword:

```
D SUMMARIZE TABLE 'https://blobs.duckdb.org/data/Star_Trek-Season_1.csv';
┌─────────────────────────────┬─────────────┬────────────┬────────────┬───────────────┬─────────────────────┬─────────────────────┬─────────┬─────────┬─────────┬───────┬─────────────────┐
│         column_name         │ column_type │    min     │    max     │ approx_unique │         avg         │         std         │   q25   │   q50   │   q75   │ count │ null_percentage │
│           varchar           │   varchar   │  varchar   │  varchar   │     int64     │       varchar       │       varchar       │ varchar │ varchar │ varchar │ int64 │  decimal(9,2)   │
├─────────────────────────────┼─────────────┼────────────┼────────────┼───────────────┼─────────────────────┼─────────────────────┼─────────┼─────────┼─────────┼───────┼─────────────────┤
│ season_num                  │ BIGINT      │ 1          │ 1          │             1 │ 1.0                 │ 0.0                 │ 1       │ 1       │ 1       │    30 │            0.00 │
│ episode_num                 │ BIGINT      │ 0          │ 29         │            30 │ 14.5                │ 8.803408430829505   │ 7       │ 14      │ 22      │    30 │            0.00 │
│ aired_date                  │ DATE        │ 1965-02-28 │ 1967-04-13 │            30 │                     │                     │         │         │         │    30 │            0.00 │
│ cnt_kirk_hookups            │ BIGINT      │ 0          │ 2          │             3 │ 0.3333333333333333  │ 0.6064784348631228  │ 0       │ 0       │ 1       │    30 │            0.00 │
│ cnt_downed_redshirts        │ BIGINT      │ 0          │ 4          │             5 │ 0.6666666666666666  │ 1.1841869998335197  │ 0       │ 0       │ 1       │    30 │            0.00 │
│ bool_aliens_almost_took_o…  │ BIGINT      │ 0          │ 1          │             2 │ 0.2                 │ 0.40683810217248617 │ 0       │ 0       │ 0       │    30 │            0.00 │
│ bool_aliens_almost_took_o…  │ BIGINT      │ 0          │ 1          │             2 │ 0.4                 │ 0.49827287912243984 │ 0       │ 0       │ 1       │    30 │            0.00 │
│ cnt_vulcan_nerve_pinch      │ BIGINT      │ 0          │ 2          │             3 │ 0.43333333333333335 │ 0.7279320417946062  │ 0       │ 0       │ 1       │    30 │            0.00 │
│ cnt_warp_speed_orders       │ BIGINT      │ 0          │ 5          │             5 │ 0.9333333333333333  │ 1.0806553992619585  │ 0       │ 1       │ 1       │    30 │            0.00 │
│ highest_warp_speed_issued   │ BIGINT      │ 0          │ 8          │             6 │ 2.033333333333333   │ 2.7975769482910016  │ 0       │ 1       │ 2       │    30 │            0.00 │
│ bool_hand_phasers_fired     │ BIGINT      │ 0          │ 1          │             2 │ 0.5333333333333333  │ 0.507416263404925   │ 0       │ 1       │ 1       │    30 │            0.00 │
│ bool_ship_phasers_fired     │ BIGINT      │ 0          │ 1          │             2 │ 0.16666666666666666 │ 0.3790490217894517  │ 0       │ 0       │ 0       │    30 │            0.00 │
│ bool_ship_photon_torpedos…  │ BIGINT      │ 0          │ 1          │             2 │ 0.03333333333333333 │ 0.18257418583505536 │ 0       │ 0       │ 0       │    30 │            0.00 │
│ cnt_transporter_pax         │ BIGINT      │ 0          │ 14         │            14 │ 5.433333333333334   │ 3.5202207349807635  │ 3       │ 5       │ 8       │    30 │            0.00 │
│ cnt_damn_it_jim_quote       │ BIGINT      │ 0          │ 0          │             1 │ 0.0                 │ 0.0                 │ 0       │ 0       │ 0       │    30 │            0.00 │
│ cnt_im_givin_her_all_shes…  │ BIGINT      │ 0          │ 0          │             1 │ 0.0                 │ 0.0                 │ 0       │ 0       │ 0       │    30 │            0.00 │
│ cnt_highly_illogical_quote  │ BIGINT      │ 0          │ 3          │             4 │ 0.3                 │ 0.7497125886079555  │ 0       │ 0       │ 0       │    30 │            0.00 │
│ bool_enterprise_saved_the…  │ BIGINT      │ 0          │ 1          │             2 │ 0.2                 │ 0.4068381021724863  │ 0       │ 0       │ 0       │    30 │            0.00 │
├─────────────────────────────┴─────────────┴────────────┴────────────┴───────────────┴─────────────────────┴─────────────────────┴─────────┴─────────┴─────────┴───────┴─────────────────┤
│ 18 rows                                                                                                                                                                      12 columns │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
D SUMMARIZE 'https://blobs.duckdb.org/data/Star_Trek-Season_1.csv';
Error: Parser Error: syntax error at or near "'https://blobs.duckdb.org/data/Star_Trek-Season_1.csv'"
LINE 1: SUMMARIZE 'https://blobs.duckdb.org/data/Star_Tre...
                  ^
```

So this arg is to accommodate this use case.

Reply:
Not sure I follow. Are you adding support for syntax that isn't supported in DuckDB (`SUMMARIZE 'path/to/file'`)?

Reply:
Nope, I added the arg so for the string case we can preserve the `TABLE` keyword

Reply:
Oh, I see, sqlglot has the same problem with summarize as it does with `table t`.

Reply:
```
In [1]: import sqlglot.expressions as sge, sqlglot as sg

In [2]: sg.parse_one("table t", read="duckdb")
Out[2]:
Alias(
  this=Column(
    this=Identifier(this=table, quoted=False)),
  alias=Identifier(this=t, quoted=False))
```

should really parse to something like `select * from t`, I can open another issue if that helps.

Reply:
Haven't seen the `TABLE t` syntax elsewhere, feel free to open a separate ticket if you can point me to the docs and I'll have a look later. We do support the `FROM t` syntax which is also equivalent to the star query:

```python
>>> import sqlglot
>>> sqlglot.parse_one("from t", "duckdb")
Select(
  expressions=[
    Star()],
  from=From(
    this=Table(
      this=Identifier(this=t, quoted=False))))
```

Reply:
It's very old, I think it either from Oracle or Postgres, postgres has had it for years.